### ISSUE TITLE
Feature Request: alter the tests to retry in testrun_completed_block

### ISSUE THREAD

**Issue by lyndsey-ferguson** [lyndsey-ferguson/fastlane-plugin-test_center#176]
#### Motivation Behind Feature

If a test has a higher probability of failing, it is a waste of test time to re-run tests with that test included. It would be nice to be able to control the next retrying test run of `multi_scan` to exclude certain tests, or to change the order of the tests.

#### Feature Description

The callback could return a Hash with different values. For example:

```ruby
test_run_block = lambda do |testrun_info|
  failed_test_count = testrun_info[:failed].size
  passed_test_count = testrun_info[:passing].size
  try_attempt = testrun_info[:try_count]
  batch = testrun_info[:batch]

  {
    continue: true,
    only_testing: testrun_info[:failed] - ['AtomicBoyUITests/AtomicBoyUITests/testExample18']
  }
end
```

Then, when the callback returns to `multi_scan's` control, it reviews the return value to determine if it should `:continue` or it should use the `:only_testing` list of tests for the next test run.

#### Alternatives or Workarounds

I don't think that there are any other work-arounds.

**Comment by lyndsey-ferguson** [comment]
Note: I've had one person requesting this behavior.

**Comment by lyndsey-ferguson** [comment]
Hello @pudupa1, I'm tagging you on this feature request as I believe that it is what you want: to alter what is re-run on a subsequent run by testing the entire test class (testsuite). Is that correct?

> @lyndsey-ferguson just stumbled across this solution! I know this is not the place to ask this. I was planning to build the similar solution now I wont. One thing I would want is re run the entire class as we have done some setup related code in the first test case to save time in automation. Would this be possible with your solution? I am really sorry for an our of context comment here!

**Comment by lyndsey-ferguson** [comment]
Note: this is the second person requesting this behavior.

**Comment by lyndsey-ferguson** [comment]
Note: @pudupa1, for your case there could be a work around performed by editing the junit file given by the [`testrun_completed_block[:report_filepath]`](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/docs/feature_details/multi_scan.md#parameters) and removing all entries for the given testsuite.

**Comment by pudupa1** [comment]
@lyndsey-ferguson thanks for bringing me here! I meant more around the class..
So we have a suit with n classes. Each class has n test cases. When any of the tests cases fail, I was looking for a way to re run all the test cases in that class.

**Comment by lyndsey-ferguson** [comment]
The structure of a test identifier is 'test target/test suite/test case'.  What you can do is use the failed tests that is given in the call back, pull the test suite and then remove each testcase for the matching test suite from the junit file. 

That file is what is used to determine which testcases to re-run.

**Comment by pudupa1** [comment]
sure.. ill give this a shot.. pretty new to all this.

**Comment by lyndsey-ferguson** [comment]
[Here](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/docs/feature_details/multi_scan.md#examples) is an example of the callback (it is not pulling the `:report_filepath` from the Hash (think map/dictionary).

[`collate_junit_reports`](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/lib/fastlane/plugin/test_center/actions/collate_junit_reports.rb) may give you some ideas on how to read XML (Junit) files. Read the [REXML library docs](https://ruby-doc.org/stdlib-2.6.1/libdoc/rexml/rdoc/REXML/Document.html) to work ruby XML.

**Comment by lyndsey-ferguson** [comment]
[Project](https://github.com/fastlane-plugin-test-center/fastlane-plugin-test_center/projects/4) started

**Comment by lyndsey-ferguson** [comment]
Released in Sponsors-first version of [v3.15.0](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/docs/feature_details/multi_scan.md)

**Comment by lyndsey-ferguson** [comment]
Released in [v.3.14.4](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.14.4)
